### PR TITLE
[SPARK-49211][SQL][FOLLOW-UP] Support catalog in QualifiedTableName

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

Support catalog in QualifiedTableName and remove `FullQualifiedTableName`.

### Why are the changes needed?

Consolidate and remove duplicate code.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing UT

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by amaliujia** [comment]
@cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/identifiers.scala [hunk 1]
```diff
@@ -107,14 +109,29 @@ case class TableIdentifier(table: String, database: Option[String], catalog: Opt
...
+    database = database,
+    name = name)
 
-case class FullQualifiedTableName(catalog: String, database: String, name: String) {
   override def toString: String = s"$catalog.$database.$name"
 }
 
+object QualifiedTableName {
+  def apply(catalog: String, database: String, name: String): QualifiedTableName = {
+    if (catalog.equalsIgnoreCase(CatalogManager.SESSION_CATALOG_NAME)) {
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/identifiers.scala:124]: why doing this check here?

Comment (amaliujia) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/identifiers.scala]: Just updated to remove this check.

**Comment by cloud-fan** [comment]
thanks, merging to master/3.5! (Since it's a followup of a 3.5 PR)

**Pull request closed without merging**